### DISCUSSION TITLE
Are `ASM_MOV_REG_IMM_FIX_U16` and `ASM_MOV_REG_IMM_FIX_WORD` still needed?

### DISCUSSION THREAD

**Discussion by agatti** [micropython/micropython#12771] [Core Development] [answered]
I'm having some fun trying to add the RISCV native code emitter to MicroPython (only RV32I/RV32M for now) and I noticed that all other native code emitters have a concrete definition for `ASM_MOV_REG_IMM_FIX_U16` and `ASM_MOV_REG_IMM_FIX_WORD` but said macros aren't used at all (at least that's what `ripgrep` tells me).

Is there any known code outside MicroPython's source tree that uses them?  If they can be safely removed I can send a PR that cleans that bit up (and I won't implement those in my emitter, of course :)).

**Comment by jimmo** [comment, marked as answer]
@agatti Excited to hear about a RISCV emitter!

Those two macros were used until the .mpy improvements last year (I can't seem to find the exact PR but it was related to, but slightly after #8191). The relevant commits are 1fb01bd6c5dc350f3c617ca8edae8dea9e5516ae and d4d53e9e114d779523e382c4ea38f0398e880aae 

So I think it should be safe to remove them.

**Reply by agatti** [reply]
@jimmo Fair enough, I'll send a PR for that then!

Regarding RISCV, the emitter may need a fair bit of internal changes - do I find the maintainers' team on discord or is there any other way to get in touch rather than on here?  Not for secrecy but for a possibly quicker response time, that is :)

**Reply by jimmo** [reply]
Yep we're on the Discord. The `#micropython-dev` channel is the best place to ask.

**Reply by jimmo** [reply]
http://micropython.org/discord
